GameOfLife.rb

```diff
@@ -0,0 +1,58 @@
+class GameOfLife
+  attr_writer :board
+  attr_reader :width
+  attr_reader :height
+
+  def initialize(width, height)
+    @board = Array.new(width) { Array.new(height) { 0 } }
+    @width = width
+    @height = height
+  end
+
+  def is_alive(x, y)
+    sum = 0
+    for i in x - 1..x + 1
+      for k in y - 1..y + 1
+        if (i != x || k != y) && k >= 0 && i >= 0 && k < @height && i < @width
+          if @board[i][k] == 1
+            sum += 1
+          end
+        end
+      end
+    end
+    if @board[x][y] == 1 && ( sum == 2 || sum == 3)
+      1
+    elsif @board[x][y] == 0 && sum == 3
+      1
+    else
+      0
+    end
+  end
+
+  def life_cycle
+    new_cycle = Array.new(@width) { Array.new(@height) { 0 } }
+    for i in 0..@width - 1
+      for k in 0..@height - 1
+        new_cycle[i][k] = is_alive(i, k)
+      end
+    end
+    @board = new_cycle
+  end
+
+  def in_the_beginning
+    @board.map! { |row| row.map! { (rand(20) % 3) % 2 } }
+  end
+
+  def display
+    puts @board.map { |x| x.join(' ') }
+  end
+
+  def get_board
+    @board
+  end
+
+  def set_board (new_board)
+    @board = new_board
+  end
+
+end

```

Comment:
wkleję Ci tutaj output rubocopa, bo on powie najwięcej:

```
GameOfLife.rb:1:1: C: Layout/EndOfLine: Carriage return character detected.
class GameOfLife ...
^^^^^^^^^^^^^^^^
GameOfLife.rb:1:1: C: Naming/FileName: The name of this source file (GameOfLife.rb) should use snake_case.
class GameOfLife
^
GameOfLife.rb:1:1: C: Style/Documentation: Missing top-level class documentation comment.
class GameOfLife
^^^^^
GameOfLife.rb:12:3: C: Metrics/AbcSize: Assignment Branch Condition size for is_alive is too high. [26.55/15]
  def is_alive(x, y) ...
  ^^^^^^^^^^^^^^^^^^
GameOfLife.rb:12:3: C: Metrics/CyclomaticComplexity: Cyclomatic complexity for is_alive is too high. [15/6]
  def is_alive(x, y) ...
  ^^^^^^^^^^^^^^^^^^
GameOfLife.rb:12:3: C: Metrics/MethodLength: Method has too many lines. [17/10]
  def is_alive(x, y) ...
  ^^^^^^^^^^^^^^^^^^
GameOfLife.rb:12:3: C: Metrics/PerceivedComplexity: Perceived complexity for is_alive is too high. [16/7]
  def is_alive(x, y) ...
  ^^^^^^^^^^^^^^^^^^
GameOfLife.rb:12:7: C: Naming/PredicateName: Rename is_alive to alive?.
  def is_alive(x, y)
      ^^^^^^^^
GameOfLife.rb:14:5: C: Style/For: Prefer each over for.
    for i in x - 1..x + 1
    ^^^
GameOfLife.rb:15:7: C: Style/For: Prefer each over for.
      for k in y - 1..y + 1
      ^^^
GameOfLife.rb:16:9: C: Style/Next: Use next to skip iteration.
        if (i != x || k != y) && k >= 0 && i >= 0 && k < @height && i < @width
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
GameOfLife.rb:17:11: C: Metrics/BlockNesting: Avoid more than 3 levels of block nesting.
          if @board[i][k] == 1 ...
          ^^^^^^^^^^^^^^^^^^^^
GameOfLife.rb:17:11: C: Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
          if @board[i][k] == 1
          ^^
GameOfLife.rb:23:30: C: Layout/SpaceInsideParens: Space inside parentheses detected.
    if @board[x][y] == 1 && ( sum == 2 || sum == 3)
                             ^
GameOfLife.rb:25:11: C: Style/NumericPredicate: Use (@board[x][y]).zero? instead of @board[x][y] == 0.
    elsif @board[x][y] == 0 && sum == 3
          ^^^^^^^^^^^^^^^^^
GameOfLife.rb:34:5: C: Style/For: Prefer each over for.
    for i in 0..@width - 1
    ^^^
GameOfLife.rb:35:7: C: Style/For: Prefer each over for.
      for k in 0..@height - 1
      ^^^
GameOfLife.rb:47:5: W: Lint/AmbiguousBlockAssociation: Parenthesize the param @board.map { |x| x.join(' ') } to make sure that the block will be associated with the @board.map method call.
    puts @board.map { |x| x.join(' ') }
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
GameOfLife.rb:50:7: C: Naming/AccessorMethodName: Do not prefix reader method names with get_.
  def get_board
      ^^^^^^^^^
GameOfLife.rb:54:7: C: Naming/AccessorMethodName: Do not prefix writer method names with set_.
  def set_board (new_board)
      ^^^^^^^^^
GameOfLife.rb:54:16: C: Layout/SpaceAfterMethodName: Do not put a space between a method name and the opening parenthesis.
  def set_board (new_board)
               ^
GameOfLife.rb:57:1: C: Layout/EmptyLinesAroundClassBody: Extra empty line detected at class body end.
```